services/cad-service/nesting/engine.py

```
@@ -0,0 +1,166 @@
+
+        for panel in panels:
+            panel_width = panel.width + kerf
+            panel_height = panel.height + kerf
+            if cursor_x + panel_width > sheet_inner_width:
+                cursor_x = self._margin
+                cursor_y += row_height + self._margin
+                row_height = 0.0
+            if cursor_y + panel_height > sheet_height - self._margin:
+                raise ValueError(
+                    "Panel layout exceeds available sheet height; provide larger sheets or add sheet splitting."
+                )
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)